flipperdevices/flipperzero-firmware
# [FL-3618] Infrared remote button index support

# What's new

- Refactor RpcApp subsystem, simplify API, add Doxygen comments
- Add support for sending Infrared signals by index (solves the duplicate name issue)

# Verification 

- Modify the companion app so that it would send button indices instead of names (use [this protobuf version](https://github.com/flipperdevices/flipperzero-protobuf/pull/47)
- Set "Log Level" to at least "Debug" on Flipper in order to see the debug output.
- Verify that both old syntax (button names) and the new one (button indices) work properly.

# Checklist (For Reviewer)
